src/esm_runscripts/filedicts.py

```diff
@@ -363,26 +371,38 @@ def _check_source_and_target(self, source_path, target_path):
             - If the parent dir of the ``target_path`` does not exist
         """
 
-        # Types
-        source_path_type = self._path_type(source_path)
-        target_path_type = self._path_type(target_path)
-        target_path_parent_type = self._path_type(target_path.parent)
+        # TODO: return True
+        # TODO: add typing
+
+        # Types. Eg. file, dir, link, or None
+        try:
+            source_path_type = self._path_type(source_path)
+            target_path_type = self._path_type(target_path)
+        except TypeError as err:
+            print(err)
+            raise
 
         # Checks
         # ------
-        # Source exists
-        if not source_path_type:
-            raise Exception(f"Source file ``{source_path}`` does not exist!")
-        # Target exist
+        # Source does not exist
+        if source_path_type is None:
+            err_msg = f"Unable to perform file operation. Source ``{source_path}`` does not exist!"
+            raise FileNotFoundError(err_msg)
+
+        # Target already exists
+        target_exists = os.path.exists(target_path) or os.path.islink(target_path)
         if target_path_type:
-            # TODO: Change this behavior
-            raise Exception(f"File ``{target_path_type}`` already exists!")
-        # Target dir exists
-        if not target_path_parent_type:
+            err_msg = f"Unable to perform file operation. Target ``{target_path}`` already exists"
+            # TODO: ??? Change this behavior
+            raise FileExistsError(err_msg)
+
+        # Target parent directory does not exist
+        if not target_path.parent.exists():
             # TODO: we might consider creating it
```

Comment:
I wouldn't do that, it has the potential to create target directories in crazy places if the user makes a mistake on typing the paths. I would only do that in the case of `name_in_{location}` containing a folder. But that would go somewhere else.

Reply:
Comment is not from me. This is basically a guard against creating that target whose parent does not exist.

Reply:
True, this comment is mine XD sorry

Reply:
🤣 yes, that happens all the time. I think the solution is placing the name initial before the comment 😄